lib/modules/us_core_guidance/clinicalnotes_sequence.rb

```diff
@@ -52,57 +52,65 @@ class USCoreR4ClinicalNotesSequence < SequenceBase
       attr_accessor :document_attachments, :report_attachments
 
       def test_clinical_notes_document_reference(category_code)
-        # TODO: update to only skip if no patient has the information we need
-        patient_id = @instance.patient_ids.split(',').first
+        skip_if_known_not_supported(:DocumentReference, [:search])
 
-        search_params = { 'patient': patient_id, 'type': category_code }
         resource_class = 'DocumentReference'
+        patient_ids = @instance.patient_ids.split(',')
 
-        skip_if_known_not_supported(:DocumentReference, [:read])
+        resources_found = patient_ids.any? do |patient_id|
+          search_params = { 'patient': patient_id, 'type': category_code }
 
-        reply = get_resource_by_params(versioned_resource_class(resource_class), search_params)
-        assert_response_ok(reply)
-        assert_bundle_response(reply)
+          reply = get_resource_by_params(versioned_resource_class(resource_class), search_params)
+          assert_response_ok(reply)
+          assert_bundle_response(reply)
 
-        resources_found = reply&.resource&.entry&.any? { |entry| entry&.resource&.resourceType == resource_class }
+          next if reply&.resource&.entry&.none? { |entry| entry&.resource&.resourceType == resource_class }
```

Comment:
```suggestion
          next unless reply&.resource&.entry&.any? { |entry| entry&.resource&.resourceType == resource_class }
```

Reply:
I was just wondering what if entry evaluates to nil?  Probably should trigger that next?  That might not be a valid bundle, in which case this is moot, but all those safe operators made me wonder what would happen if nil is returned somewhere in the chain.  Do check my logic though if you agree (it is a little mind-bending).

Reply:
Yeah, good catch.